### ISSUE TITLE
Error in matches predictions

### ISSUE THREAD

**Issue by andony-arrieula** [kochlisGit/ProphitBet-Soccer-Bets-Predictor#71]
There are 2 errors in the match prediction section:
- the data used to make the prediction is not updated, it's the data of the previous match that is reused
- the data is passed as a list of values, but the list is not given in the same order as during the training

The match prediction is not usable then.

**Comment by andony-arrieula** [comment]
It seems it's not the data not in the good order, but the preprocessing done in preprocess_dataset which is not performed on the prediction data.

**Comment by kochlisGit** [comment]
This should happen only during Cross-Validation process and it's perfectly normal. The idea is that we randomly selected a train set and a n evaluation set, to measure the performance of the models.

However, during the training period (ONLY), it should use the first 100 matches as test and the rest of the dataset should be processed in the correct order!
**Otherwise, can you print the results of the preprocess_dataset for the evaluation data to verify this?**

**Comment by andony-arrieula** [comment]
The problem is not on the evaluation data, but on the prediction data passed to the predict match dialog.

For example, I launch my trained model on Ligue 1 French league, I select Paris SG as home team (the best team in the league) and Clermont as away team (the worst team in the league), and I enter the same odd (3.00) for all possible results, the algorithm gives me probabilities of 0.32 for Paris 0.23 a draw and 0.44 for Clermont, which is totally incoherent, and this is because the data given to the model is not processed before it was passed to the model.

**Comment by kochlisGit** [comment]
So the program grabs the features from the history tables, but does not preprocess them beforing passing them to model for prediction?

**Comment by andony-arrieula** [comment]
Exactly !

But the program also does not update the statistics with the data of the previous match !

**Comment by kochlisGit** [comment]
Thanks. I will take a look into it.

**Comment by kochlisGit** [comment]
Yeah, I think that would be the best way. The rows should be processed before returned using the scaler of them model's config (if any)

**Comment by AfricaData** [comment]
Any update on this issue - how was it solved in the end? @andony-arrieula did you do a PR or fork with the changes?
